oak_ml_transparency/runner/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use anyhow::Result;
+use chrono::Utc;
+use log::warn;
+use serde_json::{self, Value};
+use sha2::{Digest, Sha256};
+
+use std::{
+    fs::{remove_file, File},
+    io::{self, Read, Write},
+    path::PathBuf,
+    process::Command,
+};
+
+const RESULT_PATH: &str = "result.json";
+
+/// Generates a claim in the form of an intoto-statement about the model, identified by the given
+/// name and digest, and the given evaluation result.
+pub fn generate_claim(
+    model_name: &str,
+    model_digest: &str,
+    script_name: &str,
+    script_digest: &str,
+    result: &str,
+) -> Result<Value> {
+    // TODO(#4196): Use objects to build an instance of the claim.
+    let claim = format!(
+        r#"{{
+            "_type": "https://in-toto.io/Statement/v0.1",
+            "subject": [{{
+                "name":"{}",
+                "digest":{{"sha256":"{}"}}
+            }}],
+            "predicateType": "https://github.com/project-oak/transparent-release/schema/claim/v1",
+            "predicate": {{
+                "claimType": "https://github.com/project-oak/transparent-release/ml-eval/v0",
+                "issuedOn": "{}",
+                "claimSpec": {{
+                    "script": {{
+                        "name":"{}",
+                        "digest":{{"sha256":"{}"}}
+                    }},
+                    "result": {}
+                }}
+            }}
+        }}"#,
+        model_name,
+        model_digest,
+        Utc::now().to_rfc3339(),
+        script_name,
+        script_digest,
+        result
+    );
+    Ok(serde_json::from_str(&claim)?)
+}
+
+/// Runs the given evaluation script on the given model, and returns the result as a string, or an
+/// error if the evaluation fails.
+pub fn run_evaluation(model_path: &PathBuf, eval_path: &PathBuf) -> Result<String> {
+    // Run python evaluation script
+    let output = Command::new("python3")
+        .arg(eval_path)
+        .arg("--model")
+        .arg(model_path)
+        .arg("--output")
+        .arg(RESULT_PATH)
+        .output()?;
+
+    if !output.status.success() {
+        io::stdout().write_all(&output.stdout)?;
+        io::stderr().write_all(&output.stderr)?;
+        anyhow::bail!(
+            "Running the evaluation failed with status {}",
+            output.status,
+        )
+    }
+
+    let mut file = File::open(RESULT_PATH)?;
+    let mut result = String::new();
+    file.read_to_string(&mut result)?;
+
+    // cleanup
+    if let Err(e) = remove_file(RESULT_PATH) {
+        warn!("could not remove the result file: {e:?}");
+    }
+
+    Ok(result)
+}
+
+/// Computes a SHA2-256 digest of `input` and returns it as a hex-encoded string.
```

Comment:
What is the current input format?

Reply:
Byte array, e.g., binary content of a file.